File: src/bmark_numeric.py
#!/usr/bin/env python
'''Benchmark Runner'''

from timeit import timeit


def benchmark():
    '''Code being benchmarked'''
    processed = {x: x * x for x in range(1001)}
    return processed


# Total, wallclock time for lots of runs.
print(timeit(benchmark, number=10000))

# Verify correctness, and compare with other benchmarked code.
ret = benchmark()
for k, v in sorted(ret.items()):
    print(str(k) + ',' + str(v))

File: src/bmark_numeric.pl
#!/usr/bin/env perl

use strict;
use warnings;

use Benchmark qw(:hireswallclock timeit);

sub benchmark {
    my %processed = map { $_ => $_ * $_ } 0..1000;
    return %processed;
}

# Total, wallclock time for lots of runs.
print ((timeit 10000, \&benchmark)->real, "\n");

# Verify correctness, and compare with other benchmarked code.
my %ret = benchmark;
print "$_,$ret{$_}\n" foreach sort { $a <=> $b } keys %ret;
